### ISSUE TITLE
OpenAI O1 model incompatibility

### ISSUE THREAD

**Issue by mtdinc** [BoltAI/BoltAI#161]
**Describe the bug**
I updated the Bolt for the O1 model, but it still does not appear in the selection from OpenAI. However, I can add it as a custom model. I can access O1 through my code in Python, and my account does have access. The new model doesn't support system messages, so it should not be sending system messages to the API endpoint. Since there is no way to disable the system message for the model, O1 is currently unusable with BoltAI.

**To Reproduce**
just add the O1 as a custom model. 

**Expected behavior**
Should not be sending system messages to o1 model 

**Screenshots**
<img width="609" alt="image" src="https://github.com/user-attachments/assets/ab5bb281-6320-4690-8f12-65b4c47d3f8c">

**Version (please complete the following information):**
 - macOS: [e.g. 15.1]
 - app version BoltAI 1.20.3 

**Additional context**
It is possible that the o1 will support system messages in the future. I understand that it might be a lot of work, and you might prefer to wait until o1 supports systems. However, currently, it is unusable with BoltAI.

**Comment by longseespace** [comment]
Yeah o1 doesn’t support system message and tool use at the moment. Can you refresh the model list again. Sometimes it doesn’t automatically pull the new models. 

It’s strange that you have access but it not showing up on the model list.

**Comment by mtdinc** [comment]
It is working now. It has started to appear in the list of available models in the OpenAI section in the app, and it functions flawlessly.
